src/concurrency/local_epoch.cpp

```diff
@@ -82,35 +74,37 @@ namespace concurrency {
         epoch_map_.erase(epoch_ptr->epoch_id_);
         epoch_queue_.pop();
       } else {
-        PL_ASSERT(epoch_ptr->epoch_id_ > epoch_lower_bound_);
-        epoch_lower_bound_ = epoch_ptr->epoch_id_ - 1;
+        PL_ASSERT(epoch_ptr->epoch_id_ > epoch_id_lower_bound_);
+        epoch_id_lower_bound_ = epoch_ptr->epoch_id_ - 1;
```

Comment:
The global epoch manger should also be responsible for updating local epoch. Otherwise, if there's no transaction assigned to a certain thread, its corresponding epoch manager will never be updated.